py/dml/codegen.py

```diff
@@ -2720,9 +2720,15 @@ def stmt_after(stmt, location, scope):
     delay = codegen_expression(delay, location, scope)
     old_delay_type = delay.ctype()
     if unit == 's':
+        clock = 'SIM_object_clock(&_dev->obj)'
```

Comment:
We should arguably switch to ps clock also for seconds. That however deserves a different PR, and is possibly breaking enough to warrant a compat feature.